NSS-Day-Cohort-69/tabloid-one-more-div
# Feature - Edit Posts

### Purpose
To allow the creator of a Post to edit their approved and published Posts

### Files Changed
- Created initial Post DTO used for updating a Post in new `PostUpdateDTO.cs` DTO class
- Added PUT endpoint for updating a Post in `PostController.cs`
- Added fetch for updating a Post in `postManager.js`
- Added routing for an Edit Post view that uses the existing PostForm module in `ApplicationViews.jsx`
- Updated logic and rendering to enable editing an existing Post in `PostForm.jsx`
- Added navigation logic to "Edit" button to navigate User to Edit Post view in `PostDetails.jsx`
- Moved `TagCreateDTO.cs` to proper location

### To Test
Fetch, setup, and run according to project README, then confirm the following
- Clicking the "Edit" button when in the Post Details view navigates the User to the Edit Post view
- The Edit Post view autopopulates with data from the Post being edited
- Submitting or cancelling an Edit autonavigates the User back to the Post being edited
- New Post view still renders and functions as expected

closes #39
